megamek/src/megamek/client/ui/swing/MiniMap.java

```diff
@@ -1435,37 +1423,32 @@ private void update() {
         @Override
         public void mouseReleased(MouseEvent me) {
             // Center main map on clicked area, if there was no dragging
-            if (dialog instanceof JDialog && !dragging) {
-                processMouseClick(me.getX(), me.getY(), me);
+            if (!dragging) {
+                Point mapPoint = SwingUtilities.convertPoint(dialog, me.getX(), me.getY(), MiniMap.this);
+                processMouseClick(mapPoint.x, mapPoint.y);
             }
-            // Clear up variables related to dragging
             dragging = false;
             setCursor(Cursor.getPredefinedCursor(Cursor.DEFAULT_CURSOR));
         }
-
     };
 
     MouseMotionListener mouseMotionListener = new MouseMotionAdapter() {
 
         @Override
-        public void mouseDragged(MouseEvent e) {
-            if (!dragging) {
-                dragging = true;
-                setCursor(Cursor.getPredefinedCursor(Cursor.HAND_CURSOR));
-            }
-            processMouseClick(e.getX(), e.getY(), e);
+        public void mouseDragged(MouseEvent me) {
+            Point mapPoint = SwingUtilities.convertPoint(dialog, me.getX(), me.getY(), MiniMap.this);
+            processMouseDrag(mapPoint.x, mapPoint.y, me.getModifiersEx());
         }
     };
 
     MouseWheelListener mouseWheelListener = (we) -> {
-        if (we.getWheelRotation() < 0) {
+        if (we.getWheelRotation() > 0 ^ GUIP.getMouseWheelZoomFlip()) {
```

Comment:
Zero raised to a power?

Reply:
Think that's an XOR

Reply:
You're correct, thanks

Reply:
Raising a number to a boolean, that would be interesting. Two to the truth?